test-basic/documents-data-movement-queryAll.js

```diff
@@ -31,6 +31,7 @@ const query = q.where(ctsQb.cts.directoryQuery('/test/dataMovement/requests/quer
 
 describe('data movement queryAll', function() {
     before(function (done) {
+        this.timeout(5000);
```

Comment:
Timeouts in tests are anti-patterns, as there's no guarantee a break won't occur. If the thing we want to test is "Make sure we can query all of the URIs that match a query", then testing that we can get more than 1 URI back - i.e. 2 - is typically as effective as testing that we can get 10k URIs back. You may need a few more than 2 URIs - i.e. 10 - for some of the test cases in this module, but I definitely would not do 10k, there's no reason to spend time writing and querying 10k docs when 10 should do just fine. 

While it may not be possible to remove the timeout, writing/querying 10 docs should allow the timeout to be both set fairly low and never exceeded.

Reply:
Hi Rob, yes lesser number of docs are working too. I added this since i wanted to test with larger number of docs. This timeout is allows the docs to be written to the database before being read. Mocha has a built-in timeout of 2 seconds and without timeout, the tests started reading the docs before all the docs were written and threw an error.